src/libretro/config/constants.cpp

```diff
@@ -169,4 +171,88 @@ bool MelonDsDs::config::IsFirmwareImage(const retro::dirent& file, Firmware::Fir
 
     memcpy(&header, &buffer, sizeof(buffer));
     return true;
-}
\ No newline at end of file
+}
+
+// A MAC address has 6 bytes, each with two hexadecimal characters,
+// and 5 colons (:) for separators
+constexpr int MacAddressStringSize = 2*6 + 5;
+
+std::optional<melonDS::MacAddress> MelonDsDs::config::ParseMacAddressFile(const retro::dirent &file) noexcept {
+    ZoneScopedN(TracyFunction);
+    ZoneText(file.path, strnlen(file.path, sizeof(file.path)));
+    retro::debug("Reading file {}", file.path);
+
+    if(!file.is_regular_file()) {
+        retro::debug("{} is not a regular file, it's not a mac address file", file.path);
+        return std::nullopt;
+    }
+
+    if(!string_ends_with(file.path, ".txt")) {
+        retro::debug("{} is not a mac address file, it does not end with .txt", file.path);
+        return std::nullopt;
+    }
+    if (file.size < MacAddressStringSize) {
+        retro::debug("{} is not a mac address file, it is too small", file.path);
+        return std::nullopt;
+    }
+
+    char buffer[MacAddressStringSize];
+    RFILE* stream = filestream_open(file.path, RETRO_VFS_FILE_ACCESS_READ, RETRO_VFS_FILE_ACCESS_HINT_NONE);
+
+    int64_t bytesRead = filestream_read(stream, &buffer, sizeof(buffer));
+    if (bytesRead < MacAddressStringSize) {
+        if (bytesRead < 0) {
+            retro::warn("Failed to read {}", file.path);
+        } else {
+            retro::warn("Tried to read {} bytes, ended up reading {} bytes instead", MacAddressStringSize, bytesRead);
+        }
+        return std::nullopt;
+    }
+
+    std::optional<melonDS::MacAddress> ret =  MelonDsDs::config::ParseMacAddress(std::string_view{buffer, sizeof(buffer)});
+    if (!ret.has_value()) {
+        retro::debug("Could not read the mac address from \"{}\"", std::string_view{buffer, sizeof(buffer)});
+    }
+    return ret;
+}
+
+std::optional<melonDS::MacAddress> MelonDsDs::config::ParseMacAddress(std::string_view s) noexcept {
+    // This would be 5 lines if scanf worked on string_view
```

Comment:
You want `std::sscanf(s.data(), /* the other stuff */)`. [`<regex>`](https://en.cppreference.com/w/cpp/regex/basic_regex) would work too, and would also allow the input to be a bit more forgiving.

Reply:
Using sscanf wouldn't work because string_view is not guaranteed to be null-terminated. See https://en.cppreference.com/w/cpp/string/basic_string_view/data . Will look into <regex> however.